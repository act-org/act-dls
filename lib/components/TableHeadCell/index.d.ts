/**
 * @prettier
 */
import * as React from 'react';
import { SortObject } from '~/types';
interface Props<T> {
    children: any;
    dataKey: string;
    setSortObject: (value: SortObject<T>) => void;
    sortable?: boolean;
    sortObject: SortObject<T>;
    style?: React.CSSProperties;
}
declare const TableHeadCell: <T>({ children, dataKey, setSortObject, sortable, sortObject, style, }: Props<T>) => React.ReactElement<any>;
export default TableHeadCell;
//# sourceMappingURL=index.d.ts.map