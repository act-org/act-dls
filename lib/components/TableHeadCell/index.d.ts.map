{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/TableHeadCell/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAEH,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAO/B,OAAO,EAAE,UAAU,EAAE,MAAM,SAAS,CAAC;AAOrC,MAAM,WAAW,KAAK,CAAC,CAAC;IACtB,QAAQ,EAAE,GAAG,CAAC;IACd,OAAO,EAAE,MAAM,CAAC;IAChB,aAAa,EAAE,CAAC,KAAK,EAAE,UAAU,CAAC,CAAC,CAAC,KAAK,IAAI,CAAC;IAC9C,QAAQ,CAAC,EAAE,OAAO,CAAC;IACnB,UAAU,EAAE,UAAU,CAAC,CAAC,CAAC,CAAC;IAC1B,KAAK,CAAC,EAAE,KAAK,CAAC,aAAa,CAAC;CAC7B;AAED,QAAA,MAAM,aAAa,uFAOL,KAAK,CAAC,YAAY,CAAC,GAAG,CA0FnC,CAAC;AAEF,eAAe,aAAa,CAAC"}