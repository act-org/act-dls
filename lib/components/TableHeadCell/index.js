"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var _clsx = _interopRequireDefault(require("clsx"));

var _Grid = _interopRequireDefault(require("@material-ui/core/Grid"));

var _TableCell = _interopRequireDefault(require("@material-ui/core/TableCell"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _icons = require("../../icons");

var _colors = _interopRequireDefault(require("../../constants/colors"));

var _sortDirectionTypes = _interopRequireDefault(require("../../constants/sortDirectionTypes"));

var _styles = _interopRequireDefault(require("./styles"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/**
 * @prettier
 */
var TableHeadCell = function TableHeadCell(_ref) {
  var children = _ref.children,
      dataKey = _ref.dataKey,
      setSortObject = _ref.setSortObject,
      _ref$sortable = _ref.sortable,
      sortable = _ref$sortable === void 0 ? false : _ref$sortable,
      sortObject = _ref.sortObject,
      style = _ref.style;
  var key = dataKey;
  var sortIsApplied = sortable && sortObject.dataKey === dataKey;

  var getStyleForIcon = function getStyleForIcon(direction) {
    if (!sortIsApplied || sortObject.direction !== direction) {
      return {};
    }

    return {
      color: _colors["default"].WHITE
    };
  };

  var toggleSort = function toggleSort() {
    var direction = _sortDirectionTypes["default"].ASC;

    if (sortIsApplied) {
      direction = sortObject.direction === _sortDirectionTypes["default"].ASC ? _sortDirectionTypes["default"].DESC : _sortDirectionTypes["default"].ASC;
    }

    setSortObject({
      dataKey: key,
      direction: direction
    });
  };

  var classes = (0, _styles["default"])();
  return /*#__PURE__*/React.createElement(_TableCell["default"], {
    classes: {
      root: (0, _clsx["default"])(classes.tableCellRoot, sortable && classes.tableCellRootSortable, sortIsApplied && classes.tableCellRootSortApplied)
    },
    component: "div",
    onClick: sortable ? toggleSort : undefined,
    style: style
  }, /*#__PURE__*/React.createElement(_Typography["default"], {
    classes: {
      root: classes.typographyRoot
    }
  }, children), sortable && /*#__PURE__*/React.createElement(_Grid["default"], {
    classes: {
      root: classes.sortContainerRoot
    }
  }, /*#__PURE__*/React.createElement(_icons.ChevronUp, {
    classes: {
      root: classes.sortIconRoot
    },
    onClick: function onClick() {
      setSortObject({
        dataKey: key,
        direction: _sortDirectionTypes["default"].ASC
      });
    },
    role: "button",
    style: getStyleForIcon(_sortDirectionTypes["default"].ASC),
    viewBox: "9 3 5 15"
  }), /*#__PURE__*/React.createElement(_icons.ChevronDown, {
    classes: {
      root: classes.sortIconRoot
    },
    onClick: function onClick() {
      setSortObject({
        dataKey: key,
        direction: _sortDirectionTypes["default"].DESC
      });
    },
    role: "button",
    style: getStyleForIcon(_sortDirectionTypes["default"].DESC),
    viewBox: "9 7 5 15"
  })));
};

var _default = TableHeadCell;
exports["default"] = _default;