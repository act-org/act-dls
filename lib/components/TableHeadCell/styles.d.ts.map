{"version":3,"file":"styles.d.ts","sourceRoot":"","sources":["../../../src/components/TableHeadCell/styles.ts"],"names":[],"mappings":"AAAA;;GAEG;;AAOH,wBAoCI"}