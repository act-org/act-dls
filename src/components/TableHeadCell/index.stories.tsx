/**
 * @prettier
 */

/* eslint-disable sort-keys */

import * as React from 'react';
import { noop } from 'lodash';
import { number, select, text } from '@storybook/addon-knobs';

import TableHeadCell from '.';

export const primary = (): React.ReactElement<any> => (
  <TableHeadCell
    dataKey="itemKey"
    setSortObject={noop}
    sortable
    sortObject={{
      dataKey: 'itemKey',
      direction: select(
        'Sort Direction',
        {
          Ascending: 'ASC',
          Descending: 'DESC',
        },
        'ASC',
      ),
    }}
    style={{
      width: number('Style:Width', 300),
    }}
  >
    {text('Text', 'Header')}
  </TableHeadCell>
);

export default {
  component: TableHeadCell,
  parameters: {
    info: {
      text: 'This is the primary variant of the TableBodyCell component.',
    },
  },
  title: 'Atoms/TableHeadCell',
};
