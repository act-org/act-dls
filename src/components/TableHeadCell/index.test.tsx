/**
 * @prettier
 */

import * as React from 'react';
import { noop } from 'lodash';
import renderer from 'react-test-renderer';

import TableHeadCell from '.';

describe('TableHeadCell', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <TableHeadCell
          dataKey="itemKey"
          setSortObject={noop}
          sortable
          sortObject={{
            dataKey: 'itemKey',
            direction: 'ASC',
          }}
        >
          child
        </TableHeadCell>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
