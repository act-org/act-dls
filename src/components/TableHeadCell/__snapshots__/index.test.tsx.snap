// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeadCell matches the snapshot 1`] = `
<div
  aria-sort={null}
  className="MuiTableCell-root makeStyles-tableCellRoot-3 makeStyles-tableCellRootSortable-4 makeStyles-tableCellRootSortApplied-5"
  onClick={[Function]}
  role="cell"
>
  <p
    className="MuiTypography-root makeStyles-typographyRoot-6 MuiTypography-body1"
  >
    child
  </p>
  <div
    className="MuiGrid-root makeStyles-sortContainerRoot-1"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root makeStyles-sortIconRoot-2"
      focusable="false"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "color": "#fff",
        }
      }
      viewBox="9 3 5 15"
    >
      <path
        d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root makeStyles-sortIconRoot-2"
      focusable="false"
      onClick={[Function]}
      role="button"
      style={Object {}}
      viewBox="9 7 5 15"
    >
      <path
        d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
      />
    </svg>
  </div>
</div>
`;
